x-pack/platform/plugins/shared/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -528,6 +528,8 @@ export const updateAgentPolicyHandler: FleetRequestHandler<
   logger.debug(`updating policy [${request.params.agentPolicyId}] in space [${spaceId}]`);
 
   try {
+    const requestSpaceId = spaceId;
```

Comment:
Fleet changes center around the agent policy update callback. When an agent policy gets updated, we check if its `spaceId` has changed. If it has, we test existing osquery packs (background tasks running on endpoints) that use this specific agent policy to ensure they're still available in the new space.